### PR TITLE
Server event system

### PR DESCRIPTION
**Pull request by tal5**

### CODE REVIEW

File: src/main/java/com/denizenscript/clientizen/serverevents/ServerEvent.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import com.denizenscript.clientizen.network.DataSerializer;
+
+public abstract class ServerEvent {
+
+	public boolean enabled;
+	public String id;
+
+	public void enable(DataDeserializer data) {}
+
+	public void disable() {}
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/serverevents/ServerEvent.java:13]: Are these two meant to be implemented in the subclasses? If so, mark them as `abstract` and remove their empty implementations

File: src/main/java/com/denizenscript/clientizen/serverevents/ServerEventManager.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+					event.enabled = false;
+					event.disable();
+				}
+			}
+			int size = message.readInt();
+			for (int i = 0; i < size; i++) {
+				String id = message.readString();
+				ServerEvent event = serverEvents.get(id);
+				if (event == null) {
+					Debug.echoError("Invalid event '" + id + "' received from server!");
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/serverevents/ServerEventManager.java:28]: Maybe add a "is your Clientizen version out of date?" to the end

**Review by acikek — CHANGES_REQUESTED** [review]

File: src/main/java/com/denizenscript/clientizen/serverevents/ServerEvent.java (hunk 1 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/serverevents/ServerEvent.java:13]: They have an empty implementation to make it easier/cleaner for events which don't send/receive any data (e.g. a `clientizen player opens player inventory` event probably won't need any data)

Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/serverevents/ServerEvent.java:14]: Added a comment clarifying that ^

File: src/main/java/com/denizenscript/clientizen/serverevents/ServerEventManager.java (hunk 2 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/serverevents/ServerEventManager.java]: Good idea, added

**Pull request merged**